File: Day240/countNumberOfHops.go
package main 

import (
	"fmt"
)

func  countHops(n int) int {
	a, b, c := 1, 2, 4
	if (n==1) {
		return a
	} else if (n == 2) {
		return b
	} else if (n == 3) {
		return c
	}
	d := 0
	for i:=4; i<=n; i++ {
		d = ((a+b)%1000000007+c)%1000000007
		a = b
		b = c
		c = d
	}
	return c
}

func main() {
	n := 4
	fmt.Println(countHops(n))
}

File: Day240/countNumberOfHops.java
package Day240;

public class countNumberOfHops {

    public static long countHops(int n) {
        long a  = 1, b = 2, c = 4;
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return b;
        }
        if (n == 3) {
            return c;
        }
        for (int i=4; i<=n; i++) {
            long d = ((a+b) %1000000007 + c) %1000000007;
            a = b;
            b = c;
            c = d;
        }
        return c;
    }

    public static void main(String[] args) {
        int n = 4;
        System.out.println(countHops(n));
    }
}